mods/cnc/maps/gdi09/gdi09-AI.lua

```diff
@@ -0,0 +1,201 @@
+--[[
+   Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+   This file is part of OpenRA, which is free software. It is made
+   available to you under the terms of the GNU General Public License
+   as published by the Free Software Foundation, either version 3 of
+   the License, or (at your option) any later version. For more
+   information, see COPYING.
+]]
+
+AttackPaths = { { waypoint7 }, { waypoint8 } }
+NodBase = { handofnod, nodairfield, nodrefinery, NodCYard, nodpower1, nodpower2, nodpower3, nodpower4, nodpower5, gun5, gun6, gun7, gun8, nodsilo1, nodsilo2, nodsilo3, nodsilo4, nodobelisk}
+
+PatrolProductionQueue = { }
+InfantryAttackGroup = { }
+InfantryGroupSize = 5
+InfantryProductionCooldown = DateTime.Minutes(3)
+InfantryProductionTypes = { "e1", "e1", "e1", "e3", "e3", "e4" }
+HarvesterProductionType = { "harv" }
+VehicleAttackGroup = { }
+VehicleGroupSize = 5
+VehicleProductionCooldown = DateTime.Minutes(3)
+VehicleProductionTypes = { "bggy", "bggy", "bggy", "ltnk", "ltnk", "arty" }
+StartingCash = 14000
+
+BaseRefinery = { type = "proc", pos = CPos.New(12, 25), cost = 1500 }
+BaseNuke1 = { type = "nuke", pos = CPos.New( 5, 24), cost = 500 }
+BaseNuke2 = { type = "nuke", pos = CPos.New( 3, 24), cost = 500 }
+BaseNuke3 = { type = "nuke", pos = CPos.New(16, 30), cost = 500 }
+BaseNuke4 = { type = "nuke", pos = CPos.New(14, 30), cost = 500 }
+BaseNuke5 = { type = "nuke", pos = CPos.New(12, 30), cost = 500 }
+InfantryProduction = { type = "hand", pos = CPos.New(15, 24), cost = 500 }
+VehicleProduction = { type = "afld", pos = CPos.New(3, 27), cost = 2000 }
+
+NodGuards = { Actor168, Actor169, Actor170, Actor171, Actor172, Actor181, Actor177, Actor188, Actor189, Actor190 }
+
+BaseBuildings = { BaseRefinery, BaseNuke1, BaseNuke2, BaseNuke3, BaseNuke4, InfantryProduction, VehicleProduction }
+
+BuildBuilding = function(building, cyard)
+	if CyardIsBuilding or Nod.Cash < building.cost then
```

Comment:
`CyardIsBuilding` appears to be used here before it is first assigned. Do we need a `CyardIsBuilding = false` in the definitions at the top of the file?

Reply:
Not in Lua.